From: Hilko Bengen <bengen@debian.org>
Date: Wed, 27 Aug 2014 23:57:06 +0200
Subject: Add hppa, hppa64, ppc64el architectures

---
 src/kernel.ml | 2 ++
 src/utils.ml  | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/kernel.ml b/src/kernel.ml
index 356ac4b..9b0e8a2 100644
--- a/src/kernel.ml
+++ b/src/kernel.ml
@@ -30,6 +30,8 @@ let patt_of_cpu host_cpu =
     | "ppc" | "powerpc" | "powerpc64" -> ["ppc"; "powerpc"; "powerpc64"]
     | "sparc" | "sparc64" -> ["sparc"; "sparc64"]
     | "amd64" | "x86_64" -> ["amd64"; "x86_64"]
+    | "parisc" | "parisc64" -> ["hppa"; "hppa64"]
+    | "ppc64el" -> ["powerpc64le"]
     | _ when host_cpu.[0] = 'i' && host_cpu.[2] = '8' && host_cpu.[3] = '6' -> ["?86"]
     | _ when String.length host_cpu >= 5 && String.sub host_cpu 0 5 = "armv7" ->  ["armmp"]
     | _ -> [host_cpu]
diff --git a/src/utils.ml b/src/utils.ml
index 4223be4..e3b8742 100644
--- a/src/utils.ml
+++ b/src/utils.ml
@@ -183,6 +183,8 @@ let compare_architecture a1 a2 =
     | a when string_prefix "armv6" a -> 32
     | a when string_prefix "armv7" a -> 32
     | a when string_prefix "armv8" a -> 64
+    | "hppa" | "parisc" -> 32
+    | "hppa64" | "parisc64" -> 64
     | "ppc" | "ppc32" -> 32
     | a when string_prefix "ppc64" a -> 64
     | "sparc" | "sparc32" -> 32
