From: Hilko Bengen <bengen@hilluzination.de>
Date: Sun, 24 Jul 2011 14:51:21 +0200
Subject: Don't overwrite debian/changelog

---
 configure.ac |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 46039da..587ae71 100644
--- a/configure.ac
+++ b/configure.ac
@@ -115,7 +115,6 @@ fi
 
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_FILES([Makefile
-        debian/changelog
         helper/Makefile
         lib/Makefile
         src/config.ml
-- 
