From: Hilko Bengen <bengen@debian.org>
Date: Wed, 27 Aug 2014 23:57:06 +0200
Subject: Added ppc64el

---
 src/kernel.ml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/kernel.ml b/src/kernel.ml
index 9a0af91..cba9c38 100644
--- a/src/kernel.ml
+++ b/src/kernel.ml
@@ -30,6 +30,7 @@ let patt_of_cpu host_cpu =
     | "ppc" | "powerpc" | "powerpc64" -> ["ppc"; "powerpc"; "powerpc64"]
     | "sparc" | "sparc64" -> ["sparc"; "sparc64"]
     | "amd64" | "x86_64" -> ["amd64"; "x86_64"]
+    | "ppc64el" -> ["powerpc64le"]
     | _ when host_cpu.[0] = 'i' && host_cpu.[2] = '8' && host_cpu.[3] = '6' -> ["?86"]
     | _ when String.length host_cpu >= 5 && String.sub host_cpu 0 5 = "armv7" ->  ["armmp"]
     | _ -> [host_cpu]
