From: Hilko Bengen <bengen@debian.org>
Date: Sat, 10 May 2014 12:57:35 +0200
Subject: Make test-execstack recognize SKIP_ variable

---
 tests/test-execstack.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tests/test-execstack.sh b/tests/test-execstack.sh
index a9cb5e0..a02803b 100755
--- a/tests/test-execstack.sh
+++ b/tests/test-execstack.sh
@@ -18,6 +18,11 @@
 
 set -e
 
+if [ -n "$SKIP_TEST_EXECSTACK" ]; then
+    echo "$0: test skipped because SKIP_TEST_EXECSTACK is set."
+    exit 77
+fi
+
 if scanelf --help >/dev/null 2>&1; then
     echo "using scanelf"
     scanelf -e ../src/supermin
