From: Hilko Bengen <bengen@debian.org>
Date: Fri, 26 Aug 2011 19:43:08 +0200
Subject: autogenerate

---
 configure |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index c35e9c8..2281ac7 100755
--- a/configure
+++ b/configure
@@ -19164,7 +19164,7 @@ fi
 
 ac_config_headers="$ac_config_headers config.h"
 
-ac_config_files="$ac_config_files Makefile debian/changelog helper/Makefile lib/Makefile src/config.ml src/Makefile"
+ac_config_files="$ac_config_files Makefile helper/Makefile lib/Makefile src/config.ml src/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -19942,7 +19942,6 @@ do
     "depfiles") CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
     "config.h") CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "debian/changelog") CONFIG_FILES="$CONFIG_FILES debian/changelog" ;;
     "helper/Makefile") CONFIG_FILES="$CONFIG_FILES helper/Makefile" ;;
     "lib/Makefile") CONFIG_FILES="$CONFIG_FILES lib/Makefile" ;;
     "src/config.ml") CONFIG_FILES="$CONFIG_FILES src/config.ml" ;;
-- 
