From: Hilko Bengen <bengen@debian.org>
Date: Fri, 11 Nov 2016 10:06:19 +0100
Subject: Look for dpkg* only in specified /usr/bin, /bin

---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3a0d25d..72ffb92 100644
--- a/configure.ac
+++ b/configure.ac
@@ -110,10 +110,10 @@ AC_PATH_PROG(URPMI,[urpmi],[no], [$PATH$PATH_SEPARATOR/usr/sbin])
 
 dnl For Debian handler.
 AC_PATH_PROG(APT_GET,[apt-get],[no])
-AC_PATH_PROG(DPKG,[dpkg],[no])
-AC_PATH_PROG(DPKG_DEB,[dpkg-deb],[no])
-AC_PATH_PROG(DPKG_QUERY,[dpkg-query],[no])
-AC_PATH_PROG(DPKG_DIVERT,[[dpkg-divert]],[no])
+AC_PATH_PROG(DPKG,[dpkg],[no],[/usr/bin:/bin])
+AC_PATH_PROG(DPKG_DEB,[dpkg-deb],[no],[/usr/bin:/bin])
+AC_PATH_PROG(DPKG_QUERY,[dpkg-query],[no],[/usr/bin:/bin])
+AC_PATH_PROG(DPKG_DIVERT,[[dpkg-divert]],[no],[/usr/bin:/bin])
 
 dnl For FrugalWare handler (currently disabled).
 AC_PATH_PROG(PACMAN_G2,[pacman-g2],[no])
