From: Hilko Bengen <bengen@debian.org>
Date: Thu, 25 Aug 2011 18:40:24 +0200
Subject: fix for architectures without native OCaml compiler

---
 Makefile.am |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 4765024..898a5cb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -56,18 +56,18 @@ SOURCES_ML = $(filter %.ml,$(SOURCES))
 BOBJECTS = $(SOURCES_ML:.ml=.cmo)
 XOBJECTS = $(SOURCES_ML:.ml=.cmx)
 
-if !HAVE_OCAMLOPT
-OBJECTS = $(BOBJECTS)
-else
-OBJECTS = $(XOBJECTS)
-endif
-
 OCAMLPACKAGES = -package unix,str
 OCAMLFLAGS = -warn-error CDEFLMPSUVXYZ
 
-febootstrap: $(OBJECTS)
-	$(OCAMLFIND) $(OCAMLBEST) $(OCAMLFLAGS) $(OCAMLPACKAGES) -linkpkg \
+if !HAVE_OCAMLOPT
+febootstrap: $(BOBJECTS)
+	$(OCAMLFIND) ocamlc $(OCAMLFLAGS) $(OCAMLPACKAGES) -linkpkg \
+	  $^ -o $@
+else
+febootstrap: $(XOBJECTS)
+	$(OCAMLFIND) ocamlopt $(OCAMLFLAGS) $(OCAMLPACKAGES) -linkpkg \
 	  $^ -o $@
+endif
 
 .mli.cmi:
 	$(OCAMLFIND) ocamlc $(OCAMLFLAGS) $(OCAMLPACKAGES) -c $< -o $@
-- 
